main-app/app/scripts/gameboard/gameboard-services/gameboard-service.js

```diff
@@ -0,0 +1,41 @@
+angular.module('Server.MyModule')
+    .service('FreshGame', ['PlayerService','NewGame','CoreData','ServerTurn','$state',  function (playerService, newGame, coreData, serverTurn, $state) {
```

Comment:
FreshGame, NewGame CoreData 
What do these do? What is the difference between a fresh game and a new one?
Also interested in serverTurn - why is that needed?

Finally - ALWAYS inject $whatever dependencies first - it is one of the few style rules the Angular team mandates,